### PR TITLE
E2C: Add initial empty state

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- adds a `direction` prop to `Box`
- adds the initial landing/empty state for the page
- this is mostly FAQs and a CTA button

image

**Why do we need this feature?**

- so there's some useful info when people first land on the page

**Who is this feature for?**

- anyone attempting to migrate to cloud

**Which issue(s) does this PR fix?**:

For #83180 

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      <Stack gap={2} direction="column">
+        <InfoItem
+          title={t('migrate-to-cloud.what-is-cloud.title', 'What is Grafana Cloud?')}
+          linkTitle={t('migrate-to-cloud.what-is-cloud.link-title', 'Learn about cloud features')}
+          linkHref="https://grafana.com/products/cloud"
+        >
+          <Trans i18nKey="migrate-to-cloud.what-is-cloud.body">
+            {
+              "Grafana cloud is a fully managed cloud-hosted observability platform ideal for cloud native environments. It's everything you love about Grafana without the overhead of maintaining, upgrading, and supporting an installation."
+            }
```
Comment (ashharrison90) [inline public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx:20]: is this the best way to be able to include apostrophes (e.g. `It's`) in strings? 🤔

File: public/app/features/admin/migrate-to-cloud/EmptyState/CallToAction.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
```
Comment (Clarity-89) [inline public/app/features/admin/migrate-to-cloud/EmptyState/CallToAction.tsx:1]: We have a `CallToActionCard` component in grafana/ui, maybe that'll work with some customisation?

File: public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx (hunk 1 above)
Comment (Clarity-89) [reply public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx:20]: I think you'd be able to use `&apos;` HTML char for that.

File: public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+
+import { Box, Stack } from '@grafana/ui';
+import { t, Trans } from 'app/core/internationalization';
+
+import { InfoItem } from './InfoItem';
+
+export const InfoPane1 = () => {
```
Comment (Clarity-89) [inline public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx:8]: Wonder if this could have more descriptive name, preferably without using numbers? Maybe at least Left/RightPane to show their positions?

File: public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <Box padding={2} backgroundColor="secondary">
+      <Stack gap={2} direction="column">
```
Comment (Clarity-89) [inline public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx:11]: I think we'd rather add the `direction` prop to the Box and then we'd get rid of the extra Stack here. I think the only reason we haven't added `direction` prop to `Box` is because we haven't had any need for it yet.

File: public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx (hunk 3 above)
Comment (ashharrison90) [reply public/app/features/admin/migrate-to-cloud/EmptyState/InfoPane1.tsx:8]: haha i was trying to avoid left/right because they wrap on smaller screens 😄 but i think you're right actually, 90% of the time it's gonna be left/right

File: public/app/features/admin/migrate-to-cloud/EmptyState/CallToAction.tsx (hunk 2 above)
Comment (ashharrison90) [reply public/app/features/admin/migrate-to-cloud/EmptyState/CallToAction.tsx:1]: this one i'm not sure about cause they're so different 🤔 basically just ends up reimplementing this component and passing it as the `callToActionElement` prop which feels a bit icky 😅

Comment (Clarity-89) [reply public/app/features/admin/migrate-to-cloud/EmptyState/CallToAction.tsx:1]: Tbh I think the current component we have could use some updates instead, as I'm pretty sure it's not in line with the design system guidelines. I'll ask UXers about this. But until then I guess we'll have to use custom components.

File: packages/grafana-ui/src/components/Layout/Box/Box.story.tsx [hunk 5]
```diff
@@ -73,6 +73,7 @@ Basic.argTypes = {
   paddingBottom: SpacingTokenControl,
   paddingLeft: SpacingTokenControl,
   paddingRight: SpacingTokenControl,
+  direction: { control: 'select', options: ['row', 'row-reverse', 'column', 'column-reverse'] },
```
Comment (joshhunt) [inline packages/grafana-ui/src/components/Layout/Box/Box.story.tsx:76]: I think we should move these changes out into a separate PR ❤️

**Review by joshhunt — APPROVED** [review]
